static/js/click_handlers.js

```diff
@@ -547,6 +548,34 @@ export function initialize() {
             onHidden: (instance) => {
                 instance.destroy();
             },
+            onShow: (instance) => {
+                // For both buddy list and top left corner pm list, `target_node`
+                // is their parent `ul` element. We canot use MutationObserver
+                // directly on the reference element because it will be removed
+                // and we need to attach it on an element which will remain in the
+                // DOM which is their parent `ul`.
+                const target_node = instance.reference.parentElement.parentElement;
+                // We only need to know if any of the `li` elements were removed.
+                const config = {attributes: false, childList: true, subtree: false};
+                const callback = function (mutationsList) {
+                    for (const mutation of mutationsList) {
+                        // Hide instance if reference is in the removed node list.
+                        if (
+                            Array.prototype.includes.call(
+                                mutation.removedNodes,
+                                instance.reference.parentElement,
+                            )
+                        ) {
+                            instance.hide();
+                        }
+                    }
+                };
+                observer = new MutationObserver(callback);
+                observer.observe(target_node, config);
+            },
+            onHide: () => {
+                observer.disconnect();
+            },
```

Comment:
How does this related to the above `onHidden` handler?
"OnHidden: Invoked once the tippy has been fully hidden and unmounted from the DOM." and onHide is "onHide: Invoked once the tippy begins to hide."  I think we need a comment explaining why we want to have some code in each if that's our intent.

(Or maybe we just want both calls in the same function?)

Reply:
Yeah, we can have them  in the same block. I don't remember why I had it in `onHide`.